### PR TITLE
tribler: fix runtime failure; libtorrent-rasterbar-2_0_x, libtorrent-rasterbar-1_2_x: migrate to pkgs/by-name

### LINKED ISSUE #325282: Tribler Unstable ModuleNotFoundError: No module named 'PyQt5.sip'
### Describe the bug
The latest version of Tribler in the Unstable branch fails to start.

```
$ tribler 
INFO:__main__:Run Tribler: Namespace(torrent='', core=False, gui_test_mode=False, allow_code_injection=False, trace_exception=False, trace_debug=False, testnet=False, chant_testnet=False, trustchain_testnet=False, tunnel_testnet=False)
INFO:__main__:Root state dir: /home/craft/.Tribler
Traceback (most recent call last):
  File "/nix/store/naj9h3xwri8l8w6qvcbsww71lgz4iwxm-tribler-7.14.0/src/run_tribler.py", line 125, in <module>
    main()
  File "/nix/store/naj9h3xwri8l8w6qvcbsww71lgz4iwxm-tribler-7.14.0/src/run_tribler.py", line 117, in main
    from tribler.gui.start_gui import run_gui
  File "/nix/store/naj9h3xwri8l8w6qvcbsww71lgz4iwxm-tribler-7.14.0/src/tribler/gui/start_gui.py", line 7, in <module>
    from PyQt5.QtCore import QSettings
ModuleNotFoundError: No module named 'PyQt5.sip'
```

### Steps To Reproduce
1. Install tribler on unstable branch
2. launch tribler in terminal

### Expected behavior
Expect Tribler to launch.

### Metadata
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.7, NixOS, 24.11 (Vicuna), 24.11pre647193.9f4128e00b0a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

### LINKED ISSUE #325293: Build failure: libtorrent-rasterbar-1_2_x
### Steps To Reproduce

1. build *libtorrent-rasterbar-1_2_x*

### Build log

```
Checking for extra build files:
checking whether example files should be built... no
checking whether test files should be built... no
checking whether python bindings should be built... yes
checking for a Python interpreter with version >= 2.4... python
checking for python... /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python
checking for python version... 3.12
checking for python platform... linux
checking for GNU default python prefix... ${prefix}
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.12/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.12/site-packages
checking for python3.12... (cached) /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python
checking for a version of Python >= '2.1.0'... yes
checking for a version of Python >= '2.4'... yes
checking for the distutils Python package... no
configure: error: cannot import Python module "distutils".
Please check your Python installation. The error was:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'distutils'
```

### Notify maintainers

unmaintained

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

- **libtorrent-rasterbar-2_0_x: migrate to pkgs/by-name**
- **libtorrent-rasterbar-1_2_x: migrate to pkgs/by-name** - Fixes #325293
- **tribler: fix runtime failure** - Fixes #325282

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>libtorrent-rasterbar-1_2_x</li>
    <li>libtorrent-rasterbar-1_2_x.dev</li>
    <li>libtorrent-rasterbar-1_2_x.python</li>
    <li>tribler</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6883,7 +6883,7 @@ self: super: with self; {
 
   libtmux = callPackage ../development/python-modules/libtmux { };
 
-  libtorrent-rasterbar = (toPythonModule (pkgs.libtorrent-rasterbar.override { inherit python; })).python;
```
Comment (mkg20001) [inline pkgs/top-level/python-packages.nix:6886]: is the python3 = python; override still necesarry?

Comment (eclairevoyant) [reply pkgs/top-level/python-packages.nix:6886]: Probably not, I can remove it.

Comment (eclairevoyant) [reply pkgs/top-level/python-packages.nix:6886]: On that note, should I also explicitly mark ` libtorrent-rasterbar-1_2_x` as disabled for 3.12 and up? (Couldn't figure out how to fix it, and the 2_0_x patch didn't do it.)

**Comment by superherointj** [comment]
I'm sorry for generating a conflict here. Only found this PR afterwards. Please rebase.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325297` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>libtorrent-rasterbar-1_2_x</li>
    <li>libtorrent-rasterbar-1_2_x.dev</li>
    <li>libtorrent-rasterbar-1_2_x.python</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325297` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>libtorrent-rasterbar-1_2_x</li>
    <li>libtorrent-rasterbar-1_2_x.dev</li>
    <li>libtorrent-rasterbar-1_2_x.python</li>
    <li>tribler</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**